substra/sdk/backends/local/backend.py

```diff
@@ -349,15 +348,18 @@ def _add_testtuple(self, spec, exist_ok, spec_options=None):
         if spec.data_manager_key is not None:
             self._db.get(schemas.Type.Dataset, spec.data_manager_key)
         if spec.test_data_sample_keys is not None:
-            [
-                self._db.get(schemas.Type.DataSample, key)
+            same_data_manager = all([
+                self._db.get(schemas.Type.DataSample, key).data_manager_key == spec.data_manager_key
```

Comment:
I believe datasamples do not have a single data_manager_key but many (they can be attached to multiple datasets).